pytables.ipynb

```diff
@@ -0,0 +1,417 @@
+    "import hetio.neo4j\n",
+    "import hetio.pathtools\n",
+    "import numpy as np\n",
+    "from scipy import sparse\n",
+    "import time\n",
+    "from hetmech.matrix import get_node_to_position, metaedge_to_adjacency_matrix"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "url = 'https://github.com/dhimmel/hetionet/raw/76550e6c93fbe92124edc71725e8c7dd4ca8b1f5/hetnet/json/hetionet-v1.0.json.bz2'\n",
+    "graph = hetio.readwrite.read_graph(url)\n",
+    "metagraph = graph.metagraph"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# NPY Saving and loading"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "r, c, gig_adj = metaedge_to_adjacency_matrix(graph, 'GiG')\n",
+    "r, c, ctd_adj = metaedge_to_adjacency_matrix(graph, 'CtD')\n",
+    "r, c, sparse_gig_adj = metaedge_to_adjacency_matrix(graph, 'GiG', sparse_threshold=1)\n",
+    "r, c, sparse_ctd_adj = metaedge_to_adjacency_matrix(graph, 'CtD', sparse_threshold=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save matrices"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### np.save"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "645.2720165252686 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Dense matrices into .npy\n",
+    "t1 = time.time()\n",
+    "np.save('data/gig_adj', gig_adj)\n",
+    "np.save('data/ctd_adj', ctd_adj)\n",
+    "t2 = time.time()\n",
+    "print(f'{(t2 - t1)*1000} ms')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "5.441904067993164 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Sparse matrices into .npy\n",
+    "t1 = time.time()\n",
+    "np.save('data/sparse_gig_adj', sparse_gig_adj)\n",
+    "np.save('data/sparse_ctd_adj', sparse_ctd_adj)\n",
+    "t2 = time.time()\n",
+    "print(f'{(t2 - t1)*1000} ms')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### sparse.save_npz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "186.43593788146973 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Sparse matrices into .npz with compression\n",
+    "t1 = time.time()\n",
+    "sparse.save_npz('data/sparse_gig_adj', sparse_gig_adj, compressed=True)\n",
+    "sparse.save_npz('data/sparse_ctd_adj', sparse_ctd_adj, compressed=True)\n",
```

Comment:
To really evaluate whether to compress or not, we'd need to know:

1. how the disk space usage compares
2. how the read-time compares. reading compressed data is sometimes faster, sometimes slower

In our case, minimizing disk space and fast reading are big priorities. However, dense arrays will likely cause most of the problems so this may not be super relevant.